templates/http/modules.j2

```diff
@@ -225,6 +236,23 @@ map {{ map_data['string'] }} {{ map_data['variable'] }} {
 
 {% endmacro %}
 
+{# NGINX HTTP Mirror -- ngx_http_mirror_module #}
+{% macro mirror(mirror) %}
+{% if mirror is defined and mirror['uri'] is defined %}
```

Comment:
There is no `off` option here. I would suggest something like
```
{% if mirror['uri'] is defined and mirror['uri'] is not mapping %}
{%  for uri in mirror['uri'] if is not string %}
mirror {{ 'off' if not uri else uri }};
{% else %}
mirror {{ 'off' if not mirror['uri'] else mirror['uri'] }};
{% endfor %}
{% endif %}
```
I haven't tested this block but if it doesn't work it should at least set you on the right track.

Reply:
Actually mirror's default is `off`, but I agree with your suggestion, because explicit is better than implicit, fixed. )

Reply:
Due to how inheritance works in `nginx` there are some times where it might actually be useful to set `mirror off` in a more specific context. This will also make the template work as expected when setting `mirror: false`. And like you said, it's also better to be explicit :)